### ISSUE TITLE
Spread tuples: [(x, x), (x, x), …] → [x, x, x, x, …]

### ISSUE THREAD

**Issue by atifaziz** [morelinq/MoreLINQ#939]
I propose to add a new operator that takes a sequence of tuples and spreads the elements of the tuple into the resulting sequence:

```c#
public static IEnumerable<T> Spread<T>(this IEnumerable<(T, T)> source)
{
    foreach (var (a, b) in source)
    {
        yield return a;
        yield return b;
    }
}
```

Additional overloads can be supplied for remaining multary tuple types.

Example:

```c#
using MoreLinq;

var xs =
    Enumerable.Range(1, 10)
              .Zip(MoreEnumerable.Return(0).Repeat())
           // ...above same as...
           // .Select(x => (x, 0))
              .Spread();

foreach (var x in xs)
    Console.WriteLine(x);
```

Outputs:

    1
    0
    2
    0
    3
    0
    4
    0
    5
    0
    6
    0
    7
    0
    8
    0
    9
    0
    10
    0

This would be more generally usable than what's proposed in #695.

**Comment by viceroypenguin** [comment]
I would argue this is a fancier more opinionated version of `Unfold`, so I'd rather see `Unfold` be updated to support this type of behavior. Suggestions: 
a) an overload that takes `IEnumerable<>` and effectively does `source.SelectMany(x => Unfold(x))`
b) an overload that takes `ValueTuple` and shortcuts the other parameters
c) both of the above?

**Comment by atifaziz** [comment]
@viceroypenguin Do you mean [`Unfold`](https://morelinq.github.io/3.3/ref/api/html/M_MoreLinq_MoreEnumerable_Unfold__3.htm) that's a sequence _generator_? I guess I'm not seeing how the above is a fancier version of that. Could you expand/clarify?

**Comment by viceroypenguin** [comment]
Sure, what you're describing here is basically sequence generation for a single `ValueTuple` (i.e. `(t, t) => [t, t]`), and a `Flatten` operation on the sequence of sequences. In effect, `Spread(source) => source.SelectMany(x => new { x.Item1, x.Item2, })`. 

Reversing operation: `IEnumerable<(T, T)> Unspread(IEnumerable<T> source) => source.Batch(2).Fold((x, y) => (x, y));`

**Comment by atifaziz** [comment]
Ah, you mean a generator for _inner_ sequence? So you're saying (a) is the more general version of (b)? Do you mean the below?

```c#
public static IEnumerable<U> Spread<T, U>(this IEnumerable<T> source, Func<T, (U, U)> selector)
{
    foreach (var item in source)
    {
        var (a, b) = selector(item);
        yield return a;
        yield return b;
    }
}
```

**Comment by declard** [comment]
That looks like
```csharp
public static IEnumerable<T> AsEnumerable<T>(this (T Left, T Right) pair)
{
    yield return pair.Left;
    yield return pair.Right;
}
```

**Comment by atifaziz** [comment]
@declard Not quite. That would be `(x, x) → [x, x]` whereas this is, as in the title, `[(x, x), (x, x), …] → [x, x, x, x, …]`.
